ledger/acctonline.go

```diff
@@ -656,23 +656,20 @@ func (ao *onlineAccounts) lookupOnlineAccountData(rnd basics.Round, addr basics.
 			// no such online account, return empty
 			return ledgercore.OnlineAccountData{}, err
 		}
-		// lookupOnlineHistory does not fetch the account db round because of the following observation:
-		// 1. ao.onlineAccountsCache update happens with ao.accountsMu taken below and in postCommit
-		// 2. If we started reading the history (lookupOnlineHistory)
-		//   1. before commitRound or while it is running => OK, read what is in DB and then add new entries in postCommit
-		//     * if commitRound deletes some history after, the cache has additional entries and updRound comparison gets a right value
-		//   2. after commitRound but before postCommit => OK, read full history, ignore the update from postCommit in writeFront's updRound comparison
-		//   3. after postCommit => OK, postCommit does not add new entry with writeFrontIfExist, but here all the full history is loaded
+		// lookupOnlineHistory fetches the account DB round from the acctrounds table (returned as validThrough)
+		// We load the entire history of this account to fill the onlineAccountsCache, so that the next lookup for
+		// this online account will not hit the on-disk DB.
 		persistedDataHistory, validThrough, err := ao.accountsq.lookupOnlineHistory(addr)
 		if err != nil || len(persistedDataHistory) == 0 {
 			return ledgercore.OnlineAccountData{}, err
 		}
-		// 3. If we fishied reading the history (lookupOnlineHistory)
+		// onlineAccountsCache update happens with ao.accountsMu taken below, and in postCommit
+		// After we finished reading the history (lookupOnlineHistory), either
 		//   1. db has not advanced (validThrough == currentDbRound) => OK
 		//   2. after commitRound but before postCommit (currentDeltaLen == len(ao.deltas)) => OK, the cache gets populated and postCommit updates the new entry
```

Comment:
should we include `currentDbRound >= ao.cachedDBRoundOnline` condition as well? it is part of the post-commit check

Reply:
updated